services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/LinkOperationsBolt.java

```diff
@@ -0,0 +1,166 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.nbworker.bolts;
+
+import org.openkilda.messaging.info.event.IslChangeType;
+import org.openkilda.messaging.info.event.IslInfoData;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.messaging.nbtopology.request.GetLinksRequest;
+import org.openkilda.messaging.nbtopology.request.LinksBaseRequest;
+import org.openkilda.pce.provider.Auth;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.error.AbstractException;
+
+import com.google.common.collect.Lists;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.neo4j.driver.v1.AccessMode;
+import org.neo4j.driver.v1.Driver;
+import org.neo4j.driver.v1.Record;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.neo4j.driver.v1.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class LinkOperationsBolt extends AbstractBolt {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinkOperationsBolt.class);
+
+    private Driver driver;
+    private final Auth neoAuth;
+
+    public LinkOperationsBolt(Auth neoAuth) {
+        this.neoAuth = neoAuth;
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        this.driver = neoAuth.getDriver();
+        super.prepare(stormConf, context, collector);
+    }
+
+
+    @Override
+    protected void handleInput(Tuple input) throws AbstractException {
+        LinksBaseRequest request = (LinksBaseRequest) input.getValueByField("request");
+        final String correlationId = input.getStringByField("correlationId");
+
+        LOGGER.debug("Received read request with correlationId {}", correlationId);
+        try (Session session = driver.session(request.isReadRequest() ? AccessMode.READ : AccessMode.WRITE)) {
+            List result = null;
+            if (request instanceof GetLinksRequest) {
+                result = getAllLinks(session);
+            } else {
+                LOGGER.warn("Received unsupported Links request: {}", correlationId);
+            }
+
+            LOGGER.debug("Processed read request with correlationId {}", correlationId);
+            getOutput().emit(Lists.newArrayList(result, correlationId));
```

Comment:
you must pass "input" to keep request traking by storm.
Also you should use Fields object, to construct output tuple.